lib/configgin.rb

```diff
@@ -41,19 +46,35 @@ def generate_jobs(job_configs, templates)
         exit 1
       end
 
-      jobs[job] = Job.new(bosh_spec, kube_namespace, kube_client, kube_client_stateful_set)
+      jobs[job] = Job.new(
+        spec: bosh_spec,
+        namespace: kube_namespace,
+        client: kube_client,
+        client_stateful_set: kube_client_stateful_set,
+        self_name: @self_name
+      )
     end
     jobs
   end
 
-  def set_job_metadata(jobs)
+  # Set the exported properties and their digests, and return the digests
+  def patch_job_metadata(jobs)
```

Comment:
How about a unit test for this function?

Reply:
Added in 4b955ff2103a637fd1352e2fdfec06a46fcff390